CONTRIBUTING.md

```diff
@@ -1,15 +1,149 @@
-# WIP: Internal contributing guide
-
-## Weave Gitops :heart:s your contributions
+## Weave GitOps :heart:s your contributions
 
 Thank you for taking the time to contribute to Weave Gitops!
 
-This guide **is a work in progress** but aims to cover all aspects of how to
-interact with the project and how to get involved in development as smoothly as possible.
+This guide **is a continuous work in progress**, we are always welcome to ideas on how to improve the experience working with the project. 
+
+## Requirements/tools
+
+This is a list of the tools you may need to install:
+
+* [go](https://go.dev) -- Primary compiler for the CLI.
+* [npm](https://www.npmjs.com/) -- Package manager for UI components.
+* [ginkgo](https://onsi.github.io/ginkgo/) -- A go testing framework.
+* [docker](https://www.docker.com/) -- Used for generating containers & testing kubernetes set-ups.
+* [golangci-lint](https://github.com/golangci/golangci-lint/) -- A go linter.
+* [buf](https://buf.build/) -- To generate the protobufs used by the API.
+* [reflex](https://github.com/cespare/reflex) -- A file watcher.
+* [kind](https://kind.sigs.k8s.io/) -- Run kubernetes clusters in docker for testing.
+* [lcov](https://github.com/linux-test-project/lcov) -- Used for code coverage.
+* [flux](https://fluxcd.io/) -- Continuous delivery system for kubernetes that weave-gitops enriches.
+
+Some other tools are installed automatically by the makefile for you:
+
+* [go-acc](https://github.com/ory/go-acc) -- Calculates code coverage for go.
+* [gcov2lcov](https://github.com/jandelgado/gcov2lcov) -- Converts output from go-acc to a format lcov understands.
+
+And some tools that are installed by the `tools/download-deps.sh` script:
+
+* [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) -- Run a kubernetes control plane locally for testing.
+* [tilt](https://tilt.dev/) -- Automatically build and deploy to a local cluster.
+
+## CLI/API development
+
+To set up a development environment for the gitops CLI
+
+### To use an existing environment
```

Comment:
we could tie into https://github.com/weaveworks/weave-gitops/pull/1804, for a quick turnaround